.github/actions/check-perf/action.yml

```diff
@@ -0,0 +1,43 @@
+name: Check performance
+description: Checks the performance log for failures
+
+inputs:
+  perflog:
+    description: the performance log path
+    required: true
+  failmark:
+    description: the performance failure marker
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Check for performance failures
+      env:
+        perflog: ${{ inputs.perflog }}
+        failmark: ${{ inputs.failmark }}
+      run: |
+        failed=false
+
+        echo "Checking '$perflog' for any '$failmark' instances"
+        ls -l "$perflog"
+        echo "Contents of '$perflog':"
+        cat "$perflog"        
+
+        failures=$(grep -i "$failmark" $perflog || true)
+
+        if [ -n "$failures" ]; then
+            echo "Found performance failures in '$perflog'"
+            echo "Failures:"
+            echo "$failures"
+            failed=true
```

Comment:
It may be useful later to also `echo` a GitHub error message to show up in the summary for the workflow that invokes this action. This message format lets GitHub add messages to the UI:

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-error-message

Reply:
We may want to add this to all other actions that we have. Let's follow up in a separate PR for this. Thanks!

Reply:
Cool thanks!